### ISSUE TITLE
Tinap error: ValueError: too many file descriptors in select()

### ISSUE THREAD

**Issue by acreskeyMoz** [tarekziade/tinap#14]
I had a Windows tinap process error-out with the log that follows.
`ValueError: too many file descriptors in select()`

Note: this was using the build from this commit https://github.com/tarekziade/tinap/commit/44d7ab07659b45d8c5fd63eb95b43c57685f5ec9

I'm retrying now with latest version (portmapping).

```
Connection made
192.168.86.180:443 <= 192.168.86.180:8081
Task was destroyed but it is pending!
task: <Task pending coro=<Forwarder._sconnect() running at C:\Users\Andrew\tools\tinap\tinap\tinap\forwarder.py:64>>
C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py:609: RuntimeWarning: coroutine 'Forwarder._sconnect' was never awaited
  self._ready.clear()
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
Task was destroyed but it is pending!
task: <Task pending coro=<BaseSelectorEventLoop._accept_connection2() running at C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\selector_events.py:210> wait_for=<Future finished result=None>>
Task was destroyed but it is pending!
task: <Task pending coro=<Forwarder._sconnect() running at C:\Users\Andrew\tools\tinap\tinap\tinap\forwarder.py:64>>
Task was destroyed but it is pending!
task: <Task pending coro=<BaseSelectorEventLoop._accept_connection2() running at C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\selector_events.py:210> wait_for=<Future finished result=None>>
Traceback (most recent call last):
  File "C:\Users\Andrew\tools\tinap\tinap\Scripts\tinap-script.py", line 11, in <module>
    load_entry_point('tinap', 'console_scripts', 'tinap')()
  File "C:\Users\Andrew\tools\tinap\tinap\tinap\__init__.py", line 119, in main
    loop.run_until_complete(server.wait_closed())
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 571, in run_until_complete
    self.run_forever()
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 539, in run_forever
    self._run_once()
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 1739, in _run_once
    event_list = self._selector.select(timeout)
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\selectors.py", line 323, in select
    r, w, _ = self._select(self._readers, self._writers, [], timeout)
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\selectors.py", line 314, in _select
    r, w, x = select.select(r, w, w, timeout)
ValueError: too many file descriptors in select()
Task was destroyed but it is pending!
task: <Task pending coro=<Throttler._dequeue() running at C:\Users\Andrew\tools\tinap\tinap\tinap\throttler.py:54> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x0000019B974293A8>()]>>
```

**Comment by acreskeyMoz** [comment]
Still seeing this on the latest version.
It happens when replaying https://www.cnn.com through WebPageReplay.

```
Connection made
Traceback (most recent call last):
  File "C:\Users\Andrew\tools\tinap\tinap\Scripts\tinap-script.py", line 11, in <module>
    load_entry_point('tinap', 'console_scripts', 'tinap')()
  File "C:\Users\Andrew\tools\tinap\tinap\tinap\__init__.py", line 157, in main
    loop.run_until_complete(server.wait_closed())
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 571, in run_until_complete
    self.run_forever()
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 539, in run_forever
    self._run_once()
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 1739, in _run_once
    event_list = self._selector.select(timeout)
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\selectors.py", line 323, in select
    r, w, _ = self._select(self._readers, self._writers, [], timeout)
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\selectors.py", line 314, in _select
    r, w, x = select.select(r, w, w, timeout)
ValueError: too many file descriptors in select()
Task was destroyed but it is pending!
task: <Task pending coro=<BaseEventLoop.create_connection() running at C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py:986> wait_for=<Future finished result=None> cb=[_release_waiter(<Future pendi...93FA56048>()]>)() at C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\tasks.py:366]>
Exception ignored in: <coroutine object BaseEventLoop.create_connection at 0x0000024940ADC9C8>
Traceback (most recent call last):
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 986, in create_connection
    ssl_handshake_timeout=ssl_handshake_timeout)
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 1016, in _create_connection_transport
    transport.close()
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\selector_events.py", line 650, in close
    self._loop.call_soon(self._call_connection_lost, None)
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 688, in call_soon
    self._check_closed()
  File "C:\Users\Andrew\AppData\Local\Programs\Python\Python37\Lib\asyncio\base_events.py", line 480, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
Task was destroyed but it is pending!
task: <Task pending coro=<Forwarder._sconnect() done, defined at C:\Users\Andrew\tools\tinap\tinap\tinap\forwarder.py:66> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x000002493FA56048>()]>>
Task was destroyed but it is pending!
task: <Task pending coro=<Throttler._dequeue() running at C:\Users\Andrew\tools\tinap\tinap\tinap\throttler.py:54> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x000002493FA56C18>()]>>
Task was destroyed but it is pending!
task: <Task pending coro=<Throttler._dequeue() running at C:\Users\Andrew\tools\tinap\tinap\tinap\throttler.py:54> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x000002493FA56B58>()]>>
```

**Comment by tarekziade** [comment]
Let's try to use ProactorEventLoop under windows

**Comment by acreskeyMoz** [comment]
This fix looks good, thank you.
Tt's been running a replay of https://www.cnn.com for 4 hours straight without problem.
I'm good to close as fixed.

**Comment by tarekziade** [comment]
@acreskeyMoz thanks for testing. Released 0.2 https://pypi.org/project/tinap/0.2
